### PR TITLE
Fix Error, empty output for info: 'VERSION'

### PR DESCRIPTION
**Pull request by cevich**
When building releases, the definitive canonical version of podman (or
podman-remote) is needed.  Previously this was accomplished by scraping
`version/version.go`.  However, due to tooling differences across
platforms, this has proven problematic, unreliable, and hard to
maintain.

Fix this by building and caching a small golang binary who's only purpose
is to print the version number to stdout.  This not only provides a quick
and reliable way to determine the current version, it also acts as a check
on the version API vs tooling that relies on it.

Lastly, remove several `RELEASE_*` Makefile definitions which aren't
actually used anywhere.  These were originally added a very long time
ago to serve as part of a long since retired release process.  The
remaining items, were updated to make use of the new `test/version/version`
binary on an as-required basis (i.e. not every time `make` is run).

### CODE REVIEW

**Comment by cevich** [comment]
> assuming happy tests.

It's Friday...I've lost all hope :rofl:

File: Makefile [hunk 1]
```diff
@@ -286,7 +285,7 @@ vendor-in-container:
 ###
 
 # Make sure to warn in case we're building without the systemd buildtag.
-bin/podman: .gopathok $(SOURCES) go.mod go.sum
+bin/podman: .gopathok test/version/version $(SOURCES) go.mod go.sum
```
Comment (Luap99) [inline Makefile:288]: Why do I have to build this for podman? I think this is only required for `podman-release.tar.gz` and `podman-remote-release-%.zip`

Comment (mheon) [reply Makefile:288]: 100% concur, `go build` is slow as molasses and I'd prefer not to add another one to the development critical path of `make`.

Comment (cevich) [reply Makefile:288]: > I think this is only required for podman-release.tar.gz

That tarball includes both podman and podman-remote :cry: 

> go build is slow as molasses

Oh?  At least for my local testing, I didn't even notice the additional `version` build time.  Anyway, there _was_ a reason I added it here...but now I can't remember :disappointed:   I'll try taking it off and see what breaks (if anything).

Comment (cevich) [reply Makefile]: Oh! Now I remember.  The `podman-msi` target makes use of `$(RELEASE_NUMBER)` in the target name.  That was causing issues if the binary didn't exist when the makefile runs.  I think I know how I can work around that though...

**Comment by cevich** [comment]
Update: Avoid make printing `make: test/version/version: File not found`

**Pull request merged**